### PR TITLE
Adds name fields to the G Suite user add screen

### PR DESCRIPTION
**Pull request by belcherj**
#### Changes proposed in this Pull Request

* Adds First Name and Last Name fields to Domains -> Email -> Add User

#### Bonus changes 

* Fixes a JS error that prevented adding extra email addresses 
* Fixes a minor typo on the success page

#### Testing instructions

* Checkout branch
* https://wordpress.com/domains/manage/{DOMAIN}/add-google-apps/{DOMAIN}
* Click `Add G Suite User`
* Observe Name fields
* Fill out fields
* Continue through checkout
* Observe that name is correct in G Suite

p2MSmN-6RF-p2

### CODE REVIEW

**Comment by belcherj** [comment]
The lints have failed as this component is in dire need of a refactor.  I would like to do that as a follow up PR as the code changes will be extensive.  https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/domains/domain-management/add-google-apps/add-email-addresses-card.jsx

File: client/my-sites/domains/domain-management/add-google-apps/add-email-addresses-card.jsx [hunk 1]
```diff
@@ -128,20 +133,56 @@ const AddEmailAddressesCard = createReactClass( {
 		);
 	},
 
-	allEmailAddressFieldsets() {
+	renderFieldsets() {
+		return this.state.fieldsets.map( ( _, index ) => {
```
Comment (stephanethomas) [inline client/my-sites/domains/domain-management/add-google-apps/add-email-addresses-card.jsx]: Oh, what is this `_` symbol? 😮

I suspect that's just a way to denote the fact that we don't use the current item being iterated on, right?

File: client/my-sites/domains/domain-management/add-google-apps/add-email-addresses-card.jsx [hunk 2]
```diff
@@ -128,20 +133,56 @@ const AddEmailAddressesCard = createReactClass( {
...
+					</div>
+					<div className="add-email-addresses-card__name-fieldsets">
+						{ this.nameFieldset( index ) }
+					</div>
+				</Fragment>
+			);
+		} );
+	},
+
+	nameFieldset( index ) {
```
Comment (stephanethomas) [inline client/my-sites/domains/domain-management/add-google-apps/add-email-addresses-card.jsx]: Not mandatory at all, but we could use the `render` prefix when naming this type of methods.

File: client/my-sites/domains/domain-management/style.scss [hunk 3]
```diff
@@ -665,13 +665,29 @@ ul.email-forwarding__list {
 	margin-top: 5px;
 }
 
+.add-email-addresses-card__name-fieldsets {
+	margin-top: 20px;
+
+	@include breakpoint( '>660px' ) {
+		display: flex;
+		flex-direction: row;
```
Comment (stephanethomas) [inline client/my-sites/domains/domain-management/style.scss]: I think `row` is the default value for `flex-direction`, so we shouldn't have to specify it.

**Comment by stephanethomas** [comment]
I was finally able to test it. I tried to add two new G Suite users to an existing G Suite account but it didn't work as expected: both new users were created with the wrong name. This is probably because of this checkout page that is displayed just after specifying new users' information:



This page is disturbing, I'm not sure why it's used for. It seems that the name I entered on this page was the name used by both G Suite users though.

As an aside, I was presented with the following `Thank You` page:



As you can see, it doesn't really match the action I just performed. This is obviously not part of the scope of this pull request, but I thought I would mention it anyway.

File: client/my-sites/domains/domain-management/add-google-apps/add-email-addresses-card.jsx (hunk 1 above)
Comment (belcherj) [reply client/my-sites/domains/domain-management/add-google-apps/add-email-addresses-card.jsx]: It was already in the code.  https://github.com/Automattic/wp-calypso/pull/29155/files/6453262add9bfe9bb941d81e17407bff974be99c#diff-4755603c8aafd8730708bb8f5ef5d6e7L134

I think it is just a placeholder to get the second prop, index.

File: client/my-sites/domains/domain-management/style.scss (hunk 3 above)
Comment (belcherj) [reply client/my-sites/domains/domain-management/style.scss]: done

File: client/my-sites/domains/domain-management/add-google-apps/add-email-addresses-card.jsx (hunk 2 above)
Comment (belcherj) [reply client/my-sites/domains/domain-management/add-google-apps/add-email-addresses-card.jsx]: done

**Comment by belcherj** [comment]
97c8692 passes name properly through to the cart.  Please review.

**Comment by gravityrail** [comment]
I found an error that is not caused by this change, but probably should be fixed at the same time.

When you click "+ Add another email address" it does nothing, and the underlying error is "Uncaught TypeError: Cannot read property 'setState' of undefined"

This is because `handleAddAnotherEmailAddress()` is not bound to `this`.

I will push a fix.

**Pull request merged**